node/service/src/lib.rs

```diff
@@ -823,20 +837,102 @@ where
+				// as necessary.
 				create_inherent_data_providers: move |block: H256, ()| {
 					let current_para_block = client_set_aside_for_cidp
 						.number(block)
 						.expect("Header lookup should succeed")
 						.expect("Header passed in as parent should be present in backend.");
+					// Fetch the starting mcq head from parent block storage
+					// It is not available through a runtime api, so we use the storage provider
+
+					fn storage_prefix_build(
+						module: &[u8],
+						storage: &[u8],
+					) -> sp_storage::StorageKey {
+						sp_storage::StorageKey(
+							[twox_128(module), twox_128(storage)].concat().to_vec(),
+						)
+					}
+
+					// Oh Yuck, and we don't even know for sure where the storage will be
+					// because it depends on the pallet's name in the given runtime.
+					// We might need a runtime api for this
+					let maybe_raw_starting_dmq_mqc_head = client_set_aside_for_cidp
+						.storage(
+							&BlockId::Hash(block),
+							&storage_prefix_build(b"ParachainSystem", b"LastDmqMqcHead"),
+						)
+						.expect("We should be able to read storage from the parent block...");
+
+					let starting_dmq_mqc_head =
+						if let Some(raw_data) = maybe_raw_starting_dmq_mqc_head {
+							parity_scale_codec::Decode::decode(&mut &raw_data.0[..])
+								.expect("Stored data should decode correctly")
+						} else {
+							H256::zero()
+						};
 					let author_id = author_id.clone();
+					let channel = downward_xcm_receiver.clone();
 
 					async move {
 						let time = sp_timestamp::InherentDataProvider::from_system_time();
 
+						//TODO maybe we don't need to collect into a Vec here. We could just
+						// leave it as an iterator.
+						let mut downward_messages: Vec<InboundDownwardMessage> = channel
+							.drain()
+							.map(|message| {
+								println!("{:?}", message);
+								message
+							})
+							.collect();
+
+						// Construct a simple downward transfer message to serialize and insert
+						// I believe this should be a `VersionedXcm<T::Call>`.
+						// Since it is generic over a runtime, I may have trouble constructing it here
+						// I could make a helper function in a runtime...
+						// Oh, unless that type info is only necessary for certain kinds of calls, and I
+						// can just put () for now...
+						let downward_transfer_message = xcm::VersionedXcm::<()>::V2(Xcm(vec![
+							ReserveAssetDeposited((Parent, 10000000000000).into()),
+							ClearOrigin,
+							BuyExecution {
+								fees: (Parent, 10000000000000).into(),
+								weight_limit: Limited(4_000_000_000),
+							},
+							DepositAsset {
+								assets: All.into(),
+								max_assets: 1,
+								beneficiary: MultiLocation::new(
+									0,
+									X1(AccountKey20 {
+										network: Any,
+										key: hex_literal::hex!(
+											"f24FF3a9CF04c71Dbc94D0b566f7A27B94566cac"
+										),
+									}),
+								),
+							},
+						]));
```

Comment:
We expect this message to have the following encoded bytes based on our [test on Moonbase Stagenet](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fwss.stagenet.moonbeam.gcp.purestake.run#/explorer/query/0xe9ad53a0013a42a7287447889d6da57cb7edcd0dc5278bd37c0d501cb8581f19)

0x02100104000100000b00a0724e18090a13000100000b00a0724e1809010300286bee0d01000400010300f24ff3a9cf04c71dbc94d0b566f7a27b94566cac